g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/record/controller/RecordController.java

```diff
@@ -56,10 +55,43 @@ public APIResponseDTO getRecoredModel(HttpServletRequest request){
 		
 	}
 	
+	@GetMapping(L10nI18nAPI.SOURCE_SYNC_RECORDS_APIV2)
+	public APIResponseDTO getRecoredV2Model(HttpServletRequest request) throws L10nAPIException{
+		logger.info("begin get the changed V3 record");
+		logger.info(LOGURLSTR, request.getRequestURL());
+		String record = request.getParameter("record");
+		APIResponseDTO responseDto = null;
+		List<RecordModel>  recordList = null;
+		if(record != null && record.equalsIgnoreCase("s3")) {
+			String lastModify = request.getParameter("lastModify");
+			long lastModifyTime =0;
+			try {
+				lastModifyTime = Long.parseLong(lastModify);
+			}catch(Exception e) {
+				logger.warn("parse lastModify error: {}", lastModify);
+				lastModifyTime=0;
+			}
+		    recordList =  recordService.getChangedRecordsS3(lastModifyTime);
+		    logger.info("s3records size {}",recordList.size());
+		
+		}else {
+			recordList =  recordService.getChangedRecords();
+		    for(RecordModel rm : recordList) {
+			   recordService.updateSynchSourceRecord(rm.getProduct(), rm.getVersion(), rm.getComponent(), rm.getLocale(), rm.getStatus());
+		    }
```

Comment:
This isn't good because if communication failure will cause sync info lost.

Reply:
it will save in agent side